tests/e2e/utils/auto-fsx-setup.py

```diff
@@ -0,0 +1,478 @@
+import argparse
+import boto3
+import subprocess
+import string
+import random
+import yaml
+import urllib.request
+from shutil import which
+from time import sleep
+
+
+def main():
+    header()
+
+    verify_prerequisites()
+
+    setup_iam_authorization()
+    setup_fsx_driver()
+    setup_fsx_file_system()
+    setup_fsx_provisioning()
+
+    footer()
+
+
+def write_to_file(fs_id):
+    if WRITE_TO_FILE or WRITE_TO_FILE == "True":
+        with open("fsx-config.txt", "w") as fh:
+            fh.write(fs_id)
+
+
+def rand_name(prefix):
+    """
+    Returns a random string of 10 ascii lowercase characters appended to the prefix
+    """
+    suffix = "".join(
+        random.choice(string.ascii_lowercase + string.digits) for _ in range(10)
+    )
+    return prefix + suffix
+
+
+def header():
+    print("=================================================================")
+    print("                          FSx Setup")
+
+
+def verify_prerequisites():
+    print("=================================================================")
+    print("                   Prerequisites Verification")
+    print("=================================================================")
+
+    verify_oidc_provider_prerequisite()
+    verify_eksctl_is_installed()
+    verify_kubectl_is_installed()
+
+
+def verify_oidc_provider_prerequisite():
+    print("Verifying OIDC provider...")
+
+    if is_oidc_provider_present():
+        print("OIDC provider found")
+    else:
+        raise Exception(
+            f"Prerequisite not met : No OIDC provider found for cluster '{CLUSTER_NAME}'!"
+        )
+
+
+def is_oidc_provider_present() -> bool:
+    iam_client = get_iam_client()
+    oidc_providers = iam_client.list_open_id_connect_providers()[
+        "OpenIDConnectProviderList"
+    ]
+
+    if len(oidc_providers) == 0:
+        return False
+
+    for oidc_provider in oidc_providers:
+        oidc_provider_tags = iam_client.get_open_id_connect_provider(
+            OpenIDConnectProviderArn=oidc_provider["Arn"]
+        )["Tags"]
+
+        if any(
+            tag["Key"] == "alpha.eksctl.io/cluster-name"
+            and tag["Value"] == CLUSTER_NAME
+            for tag in oidc_provider_tags
+        ):
+            return True
+
+    return False
+
+
+def get_iam_client():
```

Comment:
please remove all the copied code and re-use from existing file.

Reply:
I have removed most common code which was being duplicated. However, the script can be cleaned even further such as some of the utilities that are only needed for FSx filesystem creation like subnet etc are still in this file. We are planning a larger refactor for some of this code as discussed last week and a lot of this code is expected to change in the next iteration.